### PR TITLE
Refactor FindUnassignedTask

### PR DESCRIPTION
**Pull request by alfonsoeromero**
This should use much less memory as the processing is sequential sequence to sequence. Additionally, all the low complexity region processing and the intervals.py have been deleted (they're no longer needed, which is great).

This passes the tests, however, to merge we'd need to make sure that:

(a) the result runs reasonably fast in swissprot (this should definitely slower than the original version),
(b) the results uses less memory in swissprot (nothing is loaded into memory, thus lighter), and
(c) the output for swissprot is reasonably equal to the obtaine one (although the test passes, I would like to make sure of that).

Please, be picky in the comments.

### CODE REVIEW

File: gfam/tasks/find_unassigned/task.py [hunk 1]
```diff
@@ -27,102 +28,75 @@ def __init__(self,
...
+        self.seg_reader = None
 
         # these are moved here from get_unassigned
         self.regions = []
-        self.maximum = None
+        self.maximum = max(self.min_length,
+                           self.min_fragment_length)
+
         # to move somewhere else
         self.seqcat = {}
```
Comment (torresmateo) [inline gfam/tasks/find_unassigned/task.py]: This variable is now unused. It is however, the one that is used by other scripts (T09 in trello). I think we should remove this variable from here, and find a way to expose a better interface of `FindUnassignedApp` through this task.

File: gfam/tasks/find_unassigned/task.py [hunk 2]
```diff
@@ -27,102 +28,75 @@ def __init__(self,
         else:
             self.log = logger
         self.sequence_id_regexp = sequence_id_regexp
+        self.assignment_reader = None
+        self.seg_reader = None
 
         # these are moved here from get_unassigned
         self.regions = []
```
Comment (torresmateo) [inline gfam/tasks/find_unassigned/task.py]: `self.regions` is now unused

**Review by torresmateo — CHANGES_REQUESTED** [review]
Very small changes requested. I think the overall script is better. Although we have identified that it breaks dependant scripts.

File: gfam/tasks/find_unassigned/task.py (hunk 1 above)
Comment (alfonsoeromero) [reply gfam/tasks/find_unassigned/task.py]: Removed!

File: gfam/tasks/find_unassigned/task.py (hunk 2 above)
Comment (alfonsoeromero) [reply gfam/tasks/find_unassigned/task.py]: Removed!

**Review by torresmateo — APPROVED** [review]

File: gfam/tasks/find_unassigned/readers/random_access_seg_reader.py [hunk 3]
```diff
@@ -23,18 +18,65 @@ def __init__(self, seg_filename: str,
...
+            else:
+                fields = line.strip().split()
+                left = int(fields[0])
+                right = int(fields[2])
+
+                cache.append((current_prot_id, left, right))
+
+                if len(cache) > MAX_CACHE_SIZE:
+                    c.executemany(
+                        "insert into lcr lcr (?, ?, ?);", cache)
```
Comment (SantiNoto) [inline gfam/tasks/find_unassigned/readers/random_access_seg_reader.py]: this should be insert into lcr values

**Pull request merged**